### DISCUSSION TITLE
Cache working when using "Quarto: Run Cells Above"?

### DISCUSSION THREAD

**Discussion by werkstattcodes** [quarto-dev/quarto-cli#3697] [Q&A] [answered]
Apologies if this is a very basic question, but there's something with the chunk option "`cache: true`" which I don't understand.

When is the cache supposed to be used? Only when rendering the .qmd file (i.e. Quarto: Render HTML), or also when using the script interactively (e.g. Quarto: Run All Cells).

As far as I can tell, currently, it only uses the cache when rendering the project, but re-computes the chunk, despite `cache: true`, when triggering "Quarto: Run All Cells".

Is this how it is supposed to work? I personally would think that the cache should be used,  irrespectively of how the document/chunk is triggered (render/run). At least that's how I understand the documentation. But maybe I am missing something. Many thanks.

**Comment by mcanouil** [comment, marked as answer]
It's only at the document/project rendering.
Caching results while being in interactive mode is pretty unsafe, since you can change global variables which won't affect the code, but will affect the output, ultimately leading to wrong cache values being used.

This is how `knitr` works and I believe `jupyter-cache` does the same for `Jupyter`/`Python`.

**Reply by werkstattcodes** [reply]
Ok, many thanks. Would have been too convenient.

**Reply by muschellij2** [reply]
You can try using `targets` package to create a pipeline and then use `tar_read()` and also, interactively run `rmarkdown::render("DOCUMENT.qmd")` to compile the document, which should load all the output/cache.

**Reply by cderv** [reply]
Anyhow, any interactive use of cells inside IDE (RStudio IDE, VSCODE or others) are not a quarto CLI feature. This is based on integration support for those IDE. 

So even if something is possible, it would need to be considered at those integration level. 

Just clarifying so that feature request or later issue like this can be open in the right place.
